beacon-chain/sync/topic_families_without_subnets.go

```diff
@@ -0,0 +1,292 @@
+	l := &LightClientOptimisticUpdateTopicFamily{}
+	base := newBaseGossipsubTopicFamily(s, nse, s.validateLightClientOptimisticUpdate, noopHandler, l)
+	l.baseGossipsubTopicFamily = base
+	return l
+}
+
+func (l *LightClientOptimisticUpdateTopicFamily) Name() string {
+	return "LightClientOptimisticUpdateTopicFamily"
+}
+
+// Subscribe subscribes to the topic. Slot is ignored for this topic family.
+func (l *LightClientOptimisticUpdateTopicFamily) Subscribe() {
+	l.subscribeToTopics([]string{l.getFullTopicString()})
+}
+
+// UnsubscribeAll unsubscribes from all topics in the family.
+func (l *LightClientOptimisticUpdateTopicFamily) UnsubscribeAll() {
+	l.unsubscribeAll()
+}
+
+func (l *LightClientOptimisticUpdateTopicFamily) getFullTopicString() string {
+	return p2p.LcOptimisticToTopic(l.nse.ForkDigest)
+}
+
+// Light Client Finality Update (Altair+)
+var _ GossipsubTopicFamilyWithoutDynamicSubnets = (*LightClientFinalityUpdateTopicFamily)(nil)
+
+type LightClientFinalityUpdateTopicFamily struct {
+	*baseGossipsubTopicFamily
+}
+
+func NewLightClientFinalityUpdateTopicFamily(s *Service, nse params.NetworkScheduleEntry) *LightClientFinalityUpdateTopicFamily {
+	l := &LightClientFinalityUpdateTopicFamily{}
+	base := newBaseGossipsubTopicFamily(s, nse, s.validateLightClientFinalityUpdate, noopHandler, l)
+	l.baseGossipsubTopicFamily = base
+	return l
+}
+
+func (l *LightClientFinalityUpdateTopicFamily) Name() string {
+	return "LightClientFinalityUpdateTopicFamily"
+}
+
+// Subscribe subscribes to the topic. Slot is ignored for this topic family.
+func (l *LightClientFinalityUpdateTopicFamily) Subscribe() {
+	l.subscribeToTopics([]string{l.getFullTopicString()})
+}
+
+// UnsubscribeAll unsubscribes from all topics in the family.
+func (l *LightClientFinalityUpdateTopicFamily) UnsubscribeAll() {
+	l.unsubscribeAll()
+}
+
+func (l *LightClientFinalityUpdateTopicFamily) getFullTopicString() string {
+	return p2p.LcFinalityToTopic(l.nse.ForkDigest)
+}
+
+// BLS to Execution Change (Capella+)
+var _ GossipsubTopicFamilyWithoutDynamicSubnets = (*BlsToExecutionChangeTopicFamily)(nil)
+
+type BlsToExecutionChangeTopicFamily struct {
+	*baseGossipsubTopicFamily
+}
+
+func NewBlsToExecutionChangeTopicFamily(s *Service, nse params.NetworkScheduleEntry) *BlsToExecutionChangeTopicFamily {
+	b := &BlsToExecutionChangeTopicFamily{}
+	base := newBaseGossipsubTopicFamily(s, nse, s.validateBlsToExecutionChange, s.blsToExecutionChangeSubscriber, b)
+	b.baseGossipsubTopicFamily = base
+	return b
+}
+
+func (b *BlsToExecutionChangeTopicFamily) Name() string {
+	return "BlsToExecutionChangeTopicFamily"
+}
+
+// Subscribe subscribes to the topic. Slot is ignored for this topic family.
+func (b *BlsToExecutionChangeTopicFamily) Subscribe() {
+	b.subscribeToTopics([]string{b.getFullTopicString()})
+}
+
+// UnsubscribeAll unsubscribes from all topics in the family.
+func (b *BlsToExecutionChangeTopicFamily) UnsubscribeAll() {
+	b.unsubscribeAll()
```

Comment:
Rather than making the base method private and requiring this wrapper to delegate in all cases, why not spell the base method `UnsubscribeAll` so that embedding types can simply not implement `UnsubscribeAll`?

Reply:
Done